### PR TITLE
Ensure application/device .has and .get handle null/undefined arguments clearly & consistently

### PR DESCRIPTION
**Pull request by pimterry**
Connects to #274

In the current v6 beta both `device.has` and `device.get` throw a confusing error if you passed null or undefined. Now, device and application .has and .get (used internally by many methods) throw a clear error (e.g. "Application not found: null"), and this automatically allows `.has` to handle this and still return a boolean correctly everywhere.

This now isn't a breaking change for 5.0.0, I believe, because previously it'd return a device not found error anyway. Before v6, we used to turn the uuid into a string (literally `String(uuid)`) and then look for uuid `'null'` or `'undefined'`, and get pretty much exactly the same error but from the api itself.

### CODE REVIEW

File: lib/models/device.coffee [hunk 1]
```diff
@@ -192,40 +192,43 @@ getDeviceModel = (deps, opts) ->
...
+			else
+				pine.get
+					resource: 'device'
+					options:
+						expand: 'application'
+						filter:
+							# Handle shorter uuids by asserting
+							# that it is a substring of the device
+							# uuid starting at index zero.
+							$eq: [
```
Comment (Page-) [inline lib/models/device.coffee]: I think this can be replaced with `uuid: $startswith: uuidOrId`

File: lib/models/application.coffee [hunk 2]
```diff
@@ -108,27 +108,31 @@ getApplicationModel = (deps, opts) ->
...
-					filter:
-						app_name: nameOrId
-			.tap (applications) ->
-				if isEmpty(applications)
-					throw new errors.ResinApplicationNotFound(nameOrId)
+		Promise.try ->
+			if not nameOrId?
+				throw new errors.ResinApplicationNotFound(nameOrId)
+
+			else if isId(nameOrId)
```
Comment (emirotin) [inline lib/models/application.coffee]: NIT: this `else` can be dropped

File: lib/models/device.coffee (hunk 1 above)
Comment (pimterry) [inline lib/models/device.coffee]: Done

File: lib/models/application.coffee (hunk 2 above)
Comment (pimterry) [inline lib/models/application.coffee]: Done

**Comment by pimterry** [comment]
All done, and I've prepended a "// Generated by browserify" comment to the browser build, which makes github hide the diff there too.

**Pull request merged**